django/db/models/deletion.py

```diff
@@ -271,6 +271,56 @@ def delete(self):
         # number of objects deleted for each model label
         deleted_counter = Counter()
 
+        # caches query results of signals to avoid one hit per ['pre_', 'post_']
+        cached_pk_sets = dict()
+
+        def send_m2m_changed(model, obj, action):
+            """
+            Sends `m2m_changed` to all m2m relationships of this `model` that
+            contain at least one row in the table.
+            """
+            m2m_relations = [
+                f for f in model._meta.get_fields(include_hidden=True)
```

Comment:
What I mean is that no tests fail if I change this line to `f for f in model._meta.get_fields()`